src/docs/arch-block.dox

```diff
@@ -15,6 +15,35 @@ re-written, they are written to new locations in the file. The size of a block
 is a multiple of the allocation size which is set during creation of the
 associated WiredTiger data file see: WT_SESSION::create.
 
+@plantuml_start{block.png }
+@startuml{block.png}
+
+title Block Layout
+node {
+ json "Block" as J {
+    "Page Header [00-27]": {
+      "recno (column store)": "00-07",
+      "write_gen": "08-15",
+      "mem_size": "16-19",
+      "entries (or datalen for overflow pages)": "20-23",
+      "type": "24",
+      "flags": "25",
+      "unused (padding)": "26",
+      "version": "27"
+    },
+    "Block Header [28-40]": {
```

Comment:
Relate this to the code by mentioning types here. I think the Block Header is WT_BLOCK_HEADER.